indico/modules/categories/client/js/components/CategoryStatistics.jsx

```diff
@@ -0,0 +1,196 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import categoryStatisticsURL from 'indico-url:categories.statistics_json';
+
+import moment from 'moment';
+import React, {useMemo} from 'react';
+import PropTypes from 'prop-types';
+import {Chart} from 'react-charts';
+import {Container, Header, Segment, Statistic, Message, Loader} from 'semantic-ui-react';
+import {Param, Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
+
+import './CategoryStatistics.module.scss';
+
+export default function CategoryStatistics({categoryId, lang}) {
```

Comment:
maybe use `React.Context` to have the `lang` value accessibile from different components instead of passing it down the tree?

Reply:
Something like this?
 After this [line](https://github.com/indico/indico/pull/4334/files#diff-399df67992d6e77330d0e55fc8a1e072R24)
```
const localeLang = React.createContext(lang);
```
and then using like this here:
```
static lang = localeLang

// To use its value
this.context.value 
```

or I can use the `Context.Provider ` in the parent component.

What's your say on this?
@mic4ael

Reply:
I thought more about wrapping the parent component with a context provider.

Reply:
You create a context on the top level of some module: `export const LocaleContext = React.reateContext();`

Then in the components where you want to consume it: 
`const lang = useContext(LocaleContext);`

and to provide it you wrap it the topmost component in this:

`<LocaleContext.Provider value={lang}>...</LocaleContext.Provider>`